Cython/Compiler/Parsing.py

```diff
@@ -4019,6 +4044,437 @@ def p_cpp_class_attribute(s, ctx):
+        sign = s.sy
+        s.next()
+        if s.sy != 'IMAG':
+            s.error("Expected imaginary number")
+        else:
+            add_pos = s.position()
+            value = s.systring[:-1]
+            s.next()
+            res = ExprNodes.binop_node(
+                add_pos,
+                sign,
+                operand1=res,
+                operand2=ExprNodes.ImagNode(s.position(), value=value)
+            )
+
+    if not res and sy == 'IMAG':
+        value = s.systring[:-1]
+        s.next()
+        res = ExprNodes.ImagNode(pos, value=sign+value)
+        if sign == "-":
+            res = ExprNodes.UnaryMinusNode(sign_pos, operand=res)
+
+    if res:
+        return MatchCaseNodes.MatchValuePatternNode(pos, value=res)
+
+    if next_must_be_a_number:
+        s.error("Expected a number")
+    if sy == 'BEGIN_STRING':
+        res = p_atom_string(s)
+        # f-strings not being accepted is validated in PostParse
+        return MatchCaseNodes.MatchValuePatternNode(pos, value=res)
+    elif sy == 'IDENT':
+        # Note that p_atom_ident_constants includes NULL.
+        # This is a deliberate Cython addition to the pattern matching specification
+        result = p_atom_ident_constants(s)
+        if result:
+            return MatchCaseNodes.MatchValuePatternNode(pos, value=result, is_is_check=True)
+
+    s.error("Failed to match literal")
+
+
+def p_capture_pattern(s):
+    return MatchCaseNodes.MatchAndAssignPatternNode(
+        s.position(),
+        target=p_pattern_capture_target(s)
+    )
+
+
+def p_value_pattern(s):
+    if s.sy != "IDENT":
+        s.error("Expected identifier")
+    pos = s.position()
+    res = p_name(s, s.systring)
+    s.next()
+    if s.sy != '.':
+        s.error("Expected '.'")
+    while s.sy == '.':
+        attr_pos = s.position()
+        s.next()
+        attr = p_ident(s)
+        res = ExprNodes.AttributeNode(attr_pos, obj=res, attribute=attr)
+    if s.sy in ['(', '=']:
+        s.error("Unexpected symbol '%s'" % s.sy)
+    return MatchCaseNodes.MatchValuePatternNode(pos, value=res)
+
+
+def p_group_pattern(s):
+    s.expect("(")
+    pattern = p_pattern(s)
+    s.expect(")")
+    return pattern
+
+
+def p_sequence_pattern(s):
+    opener = s.sy
+    pos = s.position()
+    if opener in ['[', '(']:
+        closer = ']' if opener == '[' else ')'
+        s.next()
+        # maybe_sequence_pattern and open_sequence_pattern
+        patterns = []
+        if s.sy == closer:
+            s.next()
+        else:
+            while True:
+                patterns.append(p_maybe_star_pattern(s))
+                if s.sy == ",":
+                    s.next()
+                    if s.sy == closer:
+                        break
+                else:
+                    if opener == ')' and len(patterns) == 1:
+                        s.error("tuple-like pattern of length 1 must finish with ','")
+                    break
+            s.expect(closer)
```

Comment:
> BTW, you mixed up the brace in the opener comparison above. (So, this probably isn't tested.)

Now tested. I'll see if I can get the test back into CPython since they clearly don't have test either.